framework/src/config/config_reader.rs

```diff
@@ -13,297 +14,109 @@ pub const DEFAULT_NAME: &'static str = "zcsi";
-                    "Could not parse name for port",
-                )).into())
-            }
-        };
-
-        let rxd = match port_def.get("rxd") {
-            Some(&Value::Integer(rxd)) => rxd as i32,
-            None => NUM_RXD,
-            v => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse number of rx descriptors {:?}",
-                    v
-                )).into())
-            }
-        };
-
-        let txd = match port_def.get("txd") {
-            Some(&Value::Integer(txd)) => txd as i32,
-            None => NUM_TXD,
-            v => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse number of tx descriptors {:?}",
-                    v
-                )).into())
-            }
-        };
-
-        let loopback = match port_def.get("loopback") {
-            Some(&Value::Boolean(l)) => l,
-            None => false,
-            v => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse loopback spec {:?}",
-                    v
-                )).into())
-            }
-        };
-
-        let tso = match port_def.get("tso") {
-            Some(&Value::Boolean(l)) => l,
-            None => false,
-            v => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse tso spec {:?}",
-                    v
-                )).into())
-            }
-        };
-
-        let csum = match port_def.get("checksum") {
-            Some(&Value::Boolean(l)) => l,
-            None => false,
-            v => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse csum spec {:?}",
-                    v
-                )).into())
-            }
-        };
-
-        let symmetric_queue = port_def.contains_key("cores");
-        if symmetric_queue
-            && (port_def.contains_key("rx_cores") || port_def.contains_key("tx_cores"))
-        {
-            println!(
-                "cores specified along with rx_cores and/or tx_cores for port {}",
-                name
-            );
-            return Err(ErrorKind::ConfigurationError(format!(
-                "cores specified along with rx_cores and/or tx_cores \
-                 for port {}",
-                name
-            )).into());
-        }
+static DEFAULT_TOML: &'static str = r#"
+    duration = 0
+    remotes = []
+    agents = []
+
+    [dpdk]
+      name = "zcsi"
+      secondary = false
+      primary_core = 0
+      cores = [0]
+      strict = false
+      pool_size = 2047
+      cache_size = 32
+      ports = []
+"#;
+
+#[derive(Debug, Deserialize)]
+pub struct OccamConfiguration {
+    pub duration: u64,
+    pub remotes: Vec<RemoteConfiguration>,
+    pub agents: Vec<AgentConfiguration>,
+    pub dpdk: NetbricksConfiguration,
+}
```

Comment:
I am going to be the one that loves not having things as options here.  Default config values FTW every day!

Reply:
If all of the members of the struct have `Default` instances, you can derive `Default`. ☝️